ansible/inventory/multi_inventory.py

```diff
@@ -37,33 +145,57 @@ class MultiInventory(object):
 
     def __init__(self, args=None):
         # Allow args to be passed when called as a library
-        if not args:
-            self.args = {}
-        else:
+        self.args = {}
+        if args != None:
             self.args = args
 
+        self.accounts = []
+        self.cache_max_age = 1800
         self.cache_path = DEFAULT_CACHE_PATH
         self.config = None
+        self.config_file = None
+
+        # Store each individual restuls by account name in this variable
         self.all_inventory_results = {}
+        # Store end results in this variable
         self.result = {}
-        self.file_path = os.path.join(os.path.dirname(os.path.realpath(__file__)))
 
-        same_dir_config_file = os.path.join(self.file_path, CONFIG_FILE_NAME)
-        etc_dir_config_file = os.path.join(os.path.sep, 'etc', 'ansible', CONFIG_FILE_NAME)
+    def run(self):
+        '''This method checks to see if the local cache is valid for the inventory.
+
+           if the cache is valid; return cache
+           else the credentials are loaded from multi_inventory.yaml or from the env
+           and we attempt to get the inventory from the provider specified.
+        '''
+
+        # Finish loading configuration files
+        self.load_config_settings()
+
+        if self.args.get('refresh_cache', None) or \
+           not MultiInventoryUtils.is_cache_valid(self.cache_path, self.cache_max_age):
+            self.get_inventory()
```

Comment:
Now have a return value for get_inventory.  We need to keep the self.result for the library calls from awsutil.